src/Symfony/Component/RateLimiter/NoLock.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+use Symfony\Component\Lock\LockInterface;
+
+/**
+ * A non locking lock.
+ *
+ * This is used when no mutex is required for this rate limiter
+ * (e.g. when using in memory or session storage).
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @final
+ */
+class NoLock implements LockInterface
```

Comment:
Shouldn't this class be part of the Lock component?

Reply:
Yes, probably.